specs/v2-sketch.yaml

```diff
@@ -524,3 +554,52 @@ components:
           items:
             type: string
             format: uuid
+    TriggerSummary:
+      type: object
+      properties:
+        type:
+          description: The type of the trigger
+          type: string
+          enum: ["explicit", "webhook"]
+        vcs:
+          description: VCS data for the trigger
+          type: object
+          properties:
+            provider_name:
+              description: Name of the VCS provider
+              type: string
+              example: GitHub
+            origin_repository_url:
+              description: |
+                URL for the repository the trigger originated from. For fork-PR
+                pipelines this is the URL to the fork, for other pipelines the
+                `origin_` and `target_repository_url`s will be the same.
+              type: string
+              example: "https://github.com/CircleCI-Public/api-preview-docs"
+            target_repository_url:
+              description: |
+                URL for the repository the trigger targets (i.e. the repository
+                the PR will be merged into). For fork-PR pipelines this is the
+                URL to the parent repo, for other pipelines the `origin_` and
+                `target_repository_url`s will be the same.
+              type: string
+              example: "https://github.com/CircleCI-Public/api-preview-docs"
+            branch:
+              description: |
+                The branch the pipeline ran on. The HEAD commit on this branch
+                was used for the pipeline, unless the `revision` parameter was
+                also set.
+              type: string
+              example: "feature/design-new-api"
+            revision:
+              description: The revision of code that the pipeline ran.
+              type: string
+              example: f454a02b5d10fcccfd7d9dd7608a76d6493a98b4
+            tag:
```

Comment:
I'm feeling less happy about the `branch`/`revision`/`tag` fields propagating everywhere now. The ways that the fields can be combined makes it hard to come up with a generic `ref` type structure that doesn't also feel forced.

Reply:
@ndintenfass I know you've been thinking about this before, any thoughts?

The logic is essentially:
```clojure
(or (not tag)
    (and (not branch)
         (not revision)))
```

We could add a field with a `OneOf` schema that reflects these options
* a `tag` structure
  - we build that tag
* a `branch-inclusive-or-revision` structure
  - if just the branch is specified we build HEAD of that branch
  - if just the revision is specified we build the default branch but hard reset to the revision
  - if the branch and revision are specified we build the branch hard reset to the revision
* the field is omitted
  - we build HEAD of the default branch

Reply:
I talked to Nathan about this. The executive summary is that for triggering purposes we will only support specifying the tag xor branch as top-level keys in the trigger structure.

For fetching Pipelines we will return one of the tag or branch, along with the revision that the tag/branch was resolved to.